traffic_ops/traffic_ops_golang/ats/parentdotconfig.go

```diff
@@ -0,0 +1,1302 @@
+
+	qry := ""
+	if server.IsTopLevelCache() {
+		// multisite origins take all the org groups in to account
+		qry = `
+WITH parent_cachegroup_ids AS (
+  SELECT cg.id as v
+  FROM cachegroup cg
+  JOIN type on type.id = cg.type
+  WHERE type.name = '` + tc.CacheGroupOriginTypeName + `'
+)
+`
+	} else {
+		qry = `
+WITH server_cachegroup_ids AS (
+  SELECT cachegroup as v FROM server WHERE id = $2
+),
+parent_cachegroup_ids AS (
+  SELECT parent_cachegroup_id as v
+  FROM cachegroup WHERE id IN (SELECT v from server_cachegroup_ids)
+  UNION ALL
+  SELECT secondary_parent_cachegroup_id as v
+  FROM cachegroup WHERE id IN (SELECT v from server_cachegroup_ids)
+)
+`
+	}
+
+	qry += `
+SELECT
+  s.id,
+  s.host_name,
+  s.ip_address,
+  s.tcp_port,
+  s.cachegroup,
+  s.status,
+  s.type,
+  s.profile,
+  s.cdn_id,
+  stype.name as type_name,
+  s.domain_name
+FROM
+  server s
+  JOIN type stype ON s.type = stype.id
+  JOIN cachegroup cg ON cg.id = s.cachegroup
+  JOIN cdn on s.cdn_id = cdn.id
+  JOIN status st ON st.id = s.status
+WHERE
+  cg.id IN (SELECT v FROM parent_cachegroup_ids)
+  AND (stype.name = '` + tc.OriginTypeName + `' OR stype.name LIKE '` + tc.EdgeTypePrefix + `%' OR stype.name LIKE '` + tc.MidTypePrefix + `%')
+  AND (st.name = '` + string(tc.CacheStatusReported) + `' OR st.name = '` + string(tc.CacheStatusOnline) + `')
+  AND cdn.name = $1
+`
+
+	// TODO move qry, qryParams to separate funcs/consts
+	qryParams := []interface{}{}
+	if server.IsTopLevelCache() {
+		qryParams = []interface{}{server.CDN}
+	} else {
+		qryParams = []interface{}{server.CDN, server.ID}
+	}
+
+	rows, err := tx.Query(qry, qryParams...)
+	if err != nil {
+		return nil, nil, errors.New("querying: " + err.Error())
+	}
+	defer rows.Close()
+
+	cgServerIDs := []int{}
+	ss := []CGServer{}
+	for rows.Next() {
+		s := CGServer{}
+		if err := rows.Scan(&s.ServerID, &s.ServerHost, &s.ServerIP, &s.ServerPort, &s.CacheGroupID, &s.Status, &s.Type, &s.ProfileID, &s.CDN, &s.TypeName, &s.Domain); err != nil {
+			return nil, nil, errors.New("scanning: " + err.Error())
+		}
+		ss = append(ss, s)
+		cgServerIDs = append(cgServerIDs, int(s.ServerID))
+	}
+
+	cgServerDSes, err := getServerDSes(tx, cgServerIDs)
+	if err != nil {
+		return nil, nil, errors.New("getting cachegroup server deliveryservices: " + err.Error())
+	}
+
+	profileParams, err := getParentConfigServerCacheProfileParams(tx, cgServerIDs) // TODO change to take cg IDs directly?
+	if err != nil {
+		return nil, nil, errors.New("getting cachegroup server profile params: " + err.Error())
+	}
+
+	allDSMap := map[DeliveryServiceID]struct{}{}
+	for _, dses := range cgServerDSes {
+		for _, ds := range dses {
+			allDSMap[ds] = struct{}{}
+		}
+	}
+	allDSes := []int{}
+	for ds, _ := range allDSMap {
+		allDSes = append(allDSes, int(ds))
+	}
+
+	dsOrigins, err := getDSOrigins(tx, allDSes)
+	if err != nil {
+		return nil, nil, errors.New("getting deliveryservice origins: " + err.Error())
+	}
+
+	for _, s := range ss {
```

Comment:
I think `ss` is maybe used too far away from its initial declaration to be using such a vague name. I don't know what `for _, s := range ss {` is doing, and figuring it out means looking back quite a bit, and interpreting another code block.

Reply:
Changed.